src/filesystem/drive/directory_handle.rs

```diff
@@ -630,7 +639,7 @@ impl DirectoryHandle {
             let mut sealed_block = Vec::new();
 
             let (_, cids) = active_block
-                .encode(rng, &node_data_key, &self.current_key, &mut sealed_block)
+                .encode(rng, &mut sealed_block)
```

Comment:
I do really like that these changes allow pulling the crypto keys out from this rather deep encoding hole 🎉